### PR TITLE
correct  config.toml.example

### PR DESCRIPTION
**Pull request by GregoryIan**
### What problem does this PR solve? 
the configuration `check-mb4-value-in-utf8` will never take effect in `conf/config.toml.example`

### What is changed and how it works?
put `check-mb4-value-in-utf8` into right section of config file according to[ the syntax of the toml configuration file](https://github.com/toml-lang/toml#table)

### Check List 

Tests 

 - Manual test (add detailed scripts or steps below)
   * create a config file,  set `check-mb4-value-in-utf8 = false`
   * run tidb-server binary, check it takes effect

Related changes

 - Need to cherry-pick to the release branch

### CODE REVIEW

**Comment by tiancaiamao** [comment]
PTAL @winkyao

File: config/config.toml.example [hunk 1]
```diff
@@ -45,8 +45,11 @@ enable-streaming = false
 lower-case-table-names = 2
 
 # Make "kill query" behavior compatible with MySQL. It's not recommend to
-# turn on this option when TiDB server is behind a proxy.
-compatible-kill-query = false
+# turn on this option when TiDB server is behind a proxy. (default true)
+# compatible-kill-query = true
```
Comment (jackysp) [inline config/config.toml.example]: Why change it?

**Comment by GregoryIan** [comment]
also fixed in https://github.com/pingcap/tidb/pull/9802

**Pull request closed without merging**